astropy/coordinates/tests/test_representation_methods.py

```diff
@@ -0,0 +1,122 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+# TEST_UNICODE_LITERALS
+import numpy as np
+
+from ... import units as u
+from .. import SphericalRepresentation, Longitude, Latitude
+from ...tests.helper import pytest
+from ...utils.compat.numpycompat import NUMPY_LT_1_9
+
+
+class TestManipulation():
+    """Manipulation of Representation shapes.
+
+    Checking that attributes are manipulated correctly.
+
+    Even more exhaustive tests are done in time.tests.test_methods
+    """
+
+    def setup(self):
+        lon = Longitude(np.arange(0, 24, 4), u.hourangle)
+        lat = Latitude(np.arange(-90, 91, 30), u.deg)
+        # With same-sized arrays
+        self.s0 = SphericalRepresentation(
+            lon[:, np.newaxis] * np.ones(lat.shape),
+            lat * np.ones(lon.shape)[:, np.newaxis],
+            np.ones(lon.shape + lat.shape) * u.kpc)
+        # With unequal arrays -> these will be broadcast.
+        self.s1 = SphericalRepresentation(lon[:, np.newaxis], lat, 1. * u.kpc)
+        # For completeness on some tests, also a cartesian one
+        self.c0 = self.s0.to_cartesian()
+
+    def test_ravel(self):
+        s0_ravel = self.s0.ravel()
+        assert s0_ravel.shape == (self.s0.size,)
+        assert np.all(s0_ravel.lon == self.s0.lon.ravel())
+        assert np.may_share_memory(s0_ravel.lon, self.s0.lon)
+        assert np.may_share_memory(s0_ravel.lat, self.s0.lat)
+        assert np.may_share_memory(s0_ravel.distance, self.s0.distance)
+        # Since s1 was broadcast, ravel needs to make a copy.
+        s1_ravel = self.s1.ravel()
+        assert s1_ravel.shape == (self.s1.size,)
+        assert np.all(s1_ravel.lon == self.s1.lon.ravel())
+        assert not np.may_share_memory(s1_ravel.lat, self.s1.lat)
```

Comment:
@StuartLittlefair - in a way, yes, since that was how I found out, e.g., that initially the initialisation of new representation instances made needless copies. For large arrays, I do think it is important that one can guarantee that, e.g., a `reshape` does not make a copy unless it is needed.